scripts/hmsl/hashicorp_vault.py

```diff
@@ -0,0 +1,182 @@
+    """Stop the Hashicorp Vault server instance."""
+    stop_cmd_ret = subprocess.run(
+        ["docker", "stop", CONTAINER_NAME], capture_output=True
+    )
+    stop_cmd_ret.check_returncode()
+
+
+def execute_container_command(cmd: str):
+    """
+    Execute the given command in the container.
+    """
+    cmd_ret = subprocess.run(
+        ["docker", "exec", CONTAINER_NAME] + cmd.split(" "),
+        capture_output=True,
+    )
+    cmd_ret.check_returncode()
+
+
+def populate_server():
+    """
+    Populate the server with a known state.
+    """
+
+    # Add a v1 kv at path secret_v1
+    execute_container_command("vault secrets enable -version=1 -path=secret_v1 kv")
+
+    # Add same secrets for both paths
+    data: dict[str, dict[str, str]] = {
+        "b2c/worker/config.env": {
+            "WORKER_KEY": "my_secret_key",
+            "DB_PASSWORD": "my_password",
+        },
+        "b2c/web_app/config.env": {
+            "SECRET_KEY": "another_secret",
+            "DB_PASSWORD": "test_test",
+        },
+        "b2c/web_app/prod/config.env": {
+            "PROD_STUFF": "test",
+        },
+        "b2b/worker/config.env": {
+            "ANOTHER_PASSWORD": "my_secret_key",
+            "SECRET": "super_secret",
+        },
+        "b2b/web_app/config.env": {
+            "TESTING": "true",
+            "TIMEOUT": "15",
+        },
+        "b2b/web_app/prod/config.env": {
+            "PROD_STUFF": "test",
+        },
+    }
+    for mount in ["secret", "secret_v1"]:
+        for secret_path, secrets in data.items():
+            execute_container_command(
+                f"vault kv put -mount={mount} {secret_path} "
+                + " ".join(
+                    (
+                        f"{secret_name}={secret_value}"
+                        for secret_name, secret_value in secrets.items()
+                    )
+                )
+            )
+
+    # Add another token without any rights on the secret and secret_v1 mounts
+    execute_container_command(
+        f"vault token create -policy=default -id {RESTRICTED_TOKEN}"
+    )
+
+
+def start_hashicorp_vault_server():
+    """Start the Hashicorp Vault server"""
+    logger.debug("Starting server...")
+
+    docker_run_ret = subprocess.run(
+        [
+            "docker",
+            "run",
+            "--cap-add=IPC_LOCK",
+            "--rm",
+            "-d",
+            f"--name={CONTAINER_NAME}",
+            "--env",
+            f"VAULT_DEV_ROOT_TOKEN_ID={ROOT_TOKEN}",
+            "--env",
+            "VAULT_ADDR=http://127.0.0.1:8200",
+            "-p",
+            "8200:8200",
+            "--health-cmd=wget -q -O - http://127.0.0.1:8200/v1/sys/health",
+            "--health-start-period=1s",
+            "--health-interval=2s",
+            "hashicorp/vault",
+        ],
+        capture_output=True,
+    )
+    if docker_run_ret.returncode == 125:
+        logger.error("it seems the server is running already.")
+        exit(1)
```

Comment:
nit:
```suggestion
        sys.exit(1)
```
Interestingly, using `exit()` like you do does work, but it calls the `exit()` function added [by the `site` module for the interpreter](https://docs.python.org/3/library/constants.html?highlight=exit#constants-added-by-the-site-module).